From 8d30858665d35c45bfffede285fcdf220b1c58c3 Mon Sep 17 00:00:00 2001
From: William Joye <wjoye@cfa.harvard.edu>
Date: Wed, 29 May 2024 16:01:57 +0000
Subject: [PATCH 2/2] fix compiler issue

---
 tclxml/docObj.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/tclxml/docObj.c b/tclxml/docObj.c
index b91ab1fee..e7bd055dc 100644
--- a/tclxml/docObj.c
+++ b/tclxml/docObj.c
@@ -121,7 +121,7 @@ TclXML_libxml2_InitDocObj(interp)
   tsdPtr->errorInfoPtr->interp = interp;
   tsdPtr->errorInfoPtr->listPtr = NULL;
   tsdPtr->errorInfoPtr->nodeHandlerProc = NULL;
-  xmlSetStructuredErrorFunc((void *) tsdPtr->errorInfoPtr, TclXML_libxml2_ErrorHandler);
+  xmlSetStructuredErrorFunc((void *) tsdPtr->errorInfoPtr, (xmlStructuredErrorFunc)TclXML_libxml2_ErrorHandler);
 
   return TCL_OK;
 }
-- 
2.45.2

